### DISCUSSION TITLE
How to concat multiple static files?

### DISCUSSION THREAD

**Discussion by kazzkiq** [brunch/brunch#1879] [Q&A]
### Description

I'm building a plugin for concatenation of markdown files (without parsing them into HTML), but when I try to compile them, they're generated individually in `/public` folder.

This is the plugin code I'm trying:

```javascript
'use strict';
class ConcatCompiler {
  compileStatic({data, path}) {
    return new Promise((resolve, reject) => {
      // Should concat all ".md" files, but is compiling them individually instead
      resolve(data);
    });
  }
}

ConcatCompiler.prototype.brunchPlugin = true;
ConcatCompiler.prototype.type = 'template';
ConcatCompiler.prototype.extension = 'md';

module.exports = ConcatCompiler;
```

I'm trying to use `'template'` as the plugin "type" because I believe this would fit better, but it generates HTML files instead of a single `.md` concatenated one.

Anyone have any idea of how can I concat multiple files into one?

**Comment by shvaikalesh** [comment]
Hey @kazzkiq, thanks for bringing this up.

Currently, there is no way to achieve this except for doing it manually in `onCompile` hook.
`.html` gets in because it is default `staticTargetExtension` value.

`compileStatic` was intended to make it possible for a single plugin to compile js runtime and static html templates (so there will be less bloat in plugin ecosystem). A better API for processing assets is coming in #1664.

However, no `joinTo` for assets yet. This is the thing we are considering. To speed this up, replying with cool use cases in this thread is appreciated.

**Comment by kazzkiq** [comment]
I think one example is the very reason I opened this issue: modularization when writing markdown files.

Instead of writing a big piece of documentation, users could split the assets into multiple files for better organization and ease of maintenance. The final single-file being generated only for production:

```
static/
| - 01-introduction.md
| - 02-first-steps.md
| - 03-concerns.md
| - 04-how-to.md
| - 05-faq.md
```
would be "compiled" to:
```
public/
| - output.md
```
The same could apply to simple static websites/SPAs where for some reason there isn't any modular framework and/or template being used. Lets say I want to make a static webpage with lots of content in it. Perhaps it would be useful if I could split the pieces of my HTML into single files, and then using Brunch to generate the final `index.html` file:

```
static/
| - 01-header.html
| - 02-hero-intro.html
| - 03-features.html
| - 04-faq.html
| - 05-clients.html
| - 06-our-team.html
| - 07-talk-to-us.html
| - 08-footer.html
```
would be "compiled" to:
```
public/
| - index.html
```

**Comment by shvaikalesh** [comment]
Another reason to have this is SVG sprites. But for SVG, we can't just join files, we should wrap them. So this should happen in plugin, not in config (for users not to copy/paste ugly xml).
